src/CMakeLists.txt

```diff
@@ -345,6 +345,28 @@ if (WITH_C_API)
     )
 endif ()
 
+# =================================================================================================
+# FORD - DOCUMENTATION GENERATION
+find_program(FORD_EXE ford
+  DOC "path to the ford executable (required to generate the documentation)"
+  )
+
+# Copy the FORD project-file into the build directory
+set(FORD_PROJECT_FILE "${CMAKE_BINARY_DIR}/DBCSR.md")
+configure_file(${CMAKE_SOURCE_DIR}/DBCSR.md "${FORD_PROJECT_FILE}")
+
+# Copy the FORD project-file into the build directory
+add_custom_target(doc
+  COMMENT "Generating API documentation and doc pages"
+  COMMAND "${FORD_EXE}" "${FORD_PROJECT_FILE}"
+  VERBATIM
+  )
+add_dependencies(doc doc_copy_tests)
+if (WITH_C_API)
+  add_dependencies(doc doc_copy_examples)
```

Comment:
Now the `CMakeLists.txt` in `src/` depends on targets defined in `examples/` and `tests/`. Do you think it would make sense to move this out to a `docs/CMakeLists.txt` instead?

Reply:
good point ...